core/src/main/java/org/geysermc/geyser/item/GeyserCustomItemData.java

```diff
@@ -162,6 +173,12 @@ public Builder renderOffsets(CustomRenderOffsets renderOffsets) {
             return this;
         }
 
+        @Override
+        public Builder tags(@NonNull Set<String> tags) {
+            this.tags = tags;
```

Comment:
If the argument is marked as NonNull, it should also be checked for nullability. I personally think that null is also fine as an alternative for `Set.of()` or `Collections.emptySet()` (so `if (tags == null) tags = Set.of()`), but outright denying null is also fine with me.

Reply:
https://github.com/GeyserMC/Geyser/pull/4041/commits/0872204b1b47e7c8f5755393ec31e778e33e2a7b allows null, but still ensures that calling .tags() returns a non-null value.